cryptonight/Cargo.toml

```diff
@@ -9,9 +9,11 @@ repository = "https://github.com/Cuprate/cuprate/tree/main/cryptonight"
 
 [dependencies]
 thiserror = "1"
-
-[build-dependencies]
-cc = "1"
-
-[dev-dependencies]
-hex = "0.4"
+sha3 = "0.11.0-pre.3"
+hex = "0.4.3"
+rand = "0.9.0-alpha.2"
+groestl = "0.10.1"
+skein = "0.1.0"
+jh = "0.1.0"
+keccak = "0.2.0-pre.0"
+digest = "0.10.7"
```

Comment:
I downgraded them. I think those "pre" and "alpha" dependencies may have been imported using RustRover instead of `cargo add`.  I moved `hex` to the `dev-dependencies` section, but it was having issues finding the `encode` method if I tried setting the version from the workspace. I was able to update the existing `thiserror` dependency to get its version from the workspace though.